packages/vertexai/src/requests/request.ts

```diff
@@ -84,7 +84,9 @@ export async function getHeaders(url: RequestUrl): Promise<Headers> {
   headers.append('Content-Type', 'application/json');
   headers.append('x-goog-api-client', getClientHeaders());
   headers.append('x-goog-api-key', url.apiSettings.apiKey);
-  headers.append('X-Firebase-AppId', url.apiSettings.appId); // Will be converted to 'X-Firebase-Appid' before it's sent in the browser.
+  if (url.apiSettings.automaticDataCollectionEnabled !== false) {
```

Comment:
Maybe we should be safe and make this fail on any falsy value in case, say, the user sets it to undefined for some reason. Sometimes that's how people unset properties. So just `if (url.apiSettings.automaticDataCollectionEnabled)`?

Reply:
good point- done